### PR TITLE
gh-119950: added a functionality to check the number of weekends

### PR DESCRIPTION
**Pull request by NobitaNoobie**
# Number of weekends count function added

This can be an important metric to calculate weekdays between 2 days. Given a holiday list, this function can be an important HR metric to calculate the number of working days between any two given dates.

* Issue: gh-119950

### CODE REVIEW

**Comment by nineteendo** [comment]
@NobitaNoobie what is the issue number? It's not `#1`.

**Comment by nineteendo** [comment]
@AlexWaygood could you link this issue I just created? #119950

File: Lib/calendar.py [hunk 1]
```diff
@@ -157,6 +157,35 @@ def weekday(year, month, day):
...
+            start_date_weekday = start_date.weekday()
+            end_date_weekday = end_date.weekday()
+            first_day_of_week_dayoftheweek = start_date_weekday
+            total_days = (end_date - start_date).days + 1 #caution
+
+            div = int(total_days / 7)
+            rem = int(total_days % 7)
+
+            num_weekends = div * 2
+
```
Comment (nineteendo) [inline Lib/calendar.py]: Remove trailing space:

```suggestion

```

**Comment by nineteendo** [comment]
@NobitaNoobie, could you please stop renaming the title? It has to be `gh-119950: <whatever>`, so for example `gh-119950: added a functionality to check the number of weekends`.

**Comment by Eclips4** [comment]
@NobitaNoobie `gh-number` should point to a number of related issue

File: Lib/calendar.py (hunk 1 above)
Comment (nineteendo) [reply Lib/calendar.py]: @NobitaNoobie, you haven't removed the trailing space. It causes the lint check to fail.
Your code should also be formatted according to https://peps.python.org/pep-0008.

Comment (nineteendo) [reply Lib/calendar.py]: Note that you can make edits to https://github.com/NobitaNoobie/cpython/tree/patch-3 and the pull request will be updated accordingly. You can commit directly on that branch.

**Comment by nineteendo** [comment]
@terryjreedy, could you add "DO-NOT-MERGE"?

**Comment by Eclips4** [comment]
As commented on the issue, this should firstly be discussed on d.p.o.

**Comment by erlend-aasland** [comment]
Closing the PR as the linked issue is closed; you can re-open the PR if the idea gets traction.

**Pull request closed without merging**